### PR TITLE
feat(applications/api): Add migrationStatus 

### PR DESCRIPTION
**Pull request by Yakitrak**
## Describe your changes

Tickets:
- https://pins-ds.atlassian.net/browse/APPLICS-152 - Add `migrationStatus` to `migrate-case` function and endpoints
- https://pins-ds.atlassian.net/browse/APPLICS-224 - Add validation error reporting to migration endpoints

Ticket has been tested via swagger because data from ODW is failing validation and therefore the whole migrate-case function cannot be tested e2e.

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/migration/migrators/nsip-project-migrator.js [hunk 1]
```diff
@@ -54,20 +55,30 @@ export const migrateNsipProjects = async (models) => {
 
 		const eventType = model.publishStatus === 'published' ? EventType.Publish : EventType.Update;
 
-		await broadcastNsipProjectEvent(nsipProject, eventType);
+		if (model.migrationStatus) {
+			console.info(`Broadcasting event for case ${model.caseId}`);
```
Comment (beejjacobs) [inline apps/api/src/server/migration/migrators/nsip-project-migrator.js:59]: Ideally we use the pino logger rather than console - so we get proper JSON formatted logs.

File: apps/functions/applications-migration/migrate-case/index.js
Comment (beejjacobs) [inline apps/functions/applications-migration/migrate-case/index.js:1]: Not sure why the change here with the for...of loop? It was much clearer in the main function and avoided the extra nesting now in `migrateCase`. Also the name of `migrateCase` is now misleading, as it migrates multiple cases.

Can we revert this change please?

**Review by beejjacobs — COMMENTED** [review]
Quite a neat little change in the end. A few little comments.

File: apps/functions/applications-migration/migrate-case/index.js
Comment (Yakitrak) [reply apps/functions/applications-migration/migrate-case/index.js:1]: I updated this so the debug file can easily call the exported name function and test multiple case references (as close to what the function would do) from the newly created `debug.js` file. I can rename more appropriately if thats helpful. Or would you still prefer to revert?

File: apps/api/src/server/migration/migrators/nsip-project-migrator.js (hunk 1 above)
Comment (Yakitrak) [reply apps/api/src/server/migration/migrators/nsip-project-migrator.js]: https://github.com/Planning-Inspectorate/back-office/pull/2409/commits/7efff32722d2f5f9900fd199edc3a2a71c637552

File: apps/functions/applications-migration/migrate-case/index.js
Comment (beejjacobs) [reply apps/functions/applications-migration/migrate-case/index.js:1]: Ah I see, that makes sense.

I think let's stick with the pattern we have elsewhere then, with a function for multiple cases, that calls a function for a single case, e.g.

`migrateCases` -> log + for loop
`migrateCase` -> migration logic

Then the debug file can call `migrateCases` (though if you want to keep it close to the function logic, you could also call the default exported function).

Comment (Yakitrak) [reply apps/functions/applications-migration/migrate-case/index.js:1]: Ah that makes sense, I have updated to follow the pattern and to be fair checking one case at a time probably the norm so I've kept it the same. Thank you!

**Review by beejjacobs — APPROVED** [review]
LGTM!

**Review by JamesP-KC — APPROVED** [review]
Nice work 👍

**Review by JamesP-KC — APPROVED** [review]
